common/src/main/java/io/netty/util/concurrent/LimitLatch.java

```diff
@@ -0,0 +1,111 @@
+package io.netty.util.concurrent;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.AbstractQueuedSynchronizer;
+
+/**
+ * Shared latch that allows the latch to be acquired a limited number of times
+ * after which all subsequent requests to acquire the latch will fail.
+ */
+public class LimitLatch {

```

Comment:
Why does this class need to be public? If this class is necessary (not sure we need more than an atomic of some sort) to share between different transport types then it should live in an `internal` package.